src/Module.zig

```diff
@@ -3115,18 +3187,31 @@ pub fn semaFile(mod: *Module, file: *Scope.File) SemaError!void {
             .owner_func = null,
         };
         defer sema.deinit();
+
+        var wip_captures = try WipCaptureScope.init(gpa, &new_decl_arena.allocator, null);
+        defer wip_captures.deinit();
+
         var block_scope: Scope.Block = .{
             .parent = null,
             .sema = &sema,
             .src_decl = new_decl,
+            .wip_capture_scope = wip_captures.scope,
             .instructions = .{},
             .inlining = null,
             .is_comptime = true,
         };
         defer block_scope.instructions.deinit(gpa);
 
         if (sema.analyzeStructDecl(new_decl, main_struct_inst, struct_obj)) |_| {
+            try wip_captures.finalize();
             new_decl.analysis = .complete;
+
+            // If an error happens resolving the fields of a struct, it will be marked
+            // invalid and a proper compile error set up. But we should still look at the
+            // other types pending resolution.
+            //const src: LazySrcLoc = .{ .node_offset = 0 };
+            //sema.resolveDeclFields(block, src, ty) catch continue;
```

Comment:
is this code supposed to be uncommented?

This looks like it was cut+pasted from `resolvePendingTypes` and I'm a bit confused about what it's doing here.

Reply:
Oops, good catch, forgot to delete this.